### ISSUE TITLE
CI on public repos started to fail with "Please, set COVERALLS_REPO_TOKEN env variable. It is required."

### ISSUE THREAD

**Issue by avodonosov** [fukamachi/cl-coveralls#14]
We use cl-coverals to test cl+ssl, and recently it started to fail with error "Please, set COVERALLS_REPO_TOKEN env variable. It is required."

The requirement is introduced by pull request #13.

Actually, for public repos the token is not required:
> The option repo_token (found on your repository’s page on Coveralls) is used to specify which project on Coveralls your project maps to. This is only needed for private repos and should be kept secret – anyone could use it to submit coverage data on your repo’s behalf.

https://docs.coveralls.io/supported-ci-services

cc @svetlyak40wt

**Comment by svetlyak40wt** [comment]
@avodonosov sorry for this issue. I've found the token is required when you are running code in Github Action. Moreover, in this case, it should be not Coveralls Token, but GitHub token :(

Coverall's Documentation is very outdated. There are two threads I've found. Both discussions are about token related issues and ambiguity of docs:

- https://github.com/lemurheavy/coveralls-public/issues/632
- https://github.com/lemurheavy/coveralls-public/issues/1508

Anyway, I'll make a fix to check if token is given only when running inside the Github Actions.

**Comment by svetlyak40wt** [comment]
Ah, and one more thing. Coveralls [API documentation](https://docs.coveralls.io/api-reference) says, that repo_token is `required` parameter. When I've tested `cl-coveralls` didn't work without it for me. Don't know how it works on Travis :-/

<img width="872" alt="Screen Shot 2021-03-08 at 09 47 41" src="https://user-images.githubusercontent.com/24827/110284838-64366900-7ff3-11eb-8ac8-ddd7d9137b5d.png">

**Comment by avodonosov** [comment]
@svetlyak40wt, were you testing github actions on a private repo or on a public repo?

**Comment by svetlyak40wt** [comment]
On a public.

пн, 8 мар. 2021 г., 14:06 Anton Vodonosov <notifications@github.com>:

**Comment by avodonosov** [comment]
@svetlyak40wt, it is strange to think that Travis CI works without token wile github actions require token.

Maybe when testing you missed something else? Have you registered and enabled your repo on the coveralls website?

**Comment by svetlyak40wt** [comment]
Yes. By the way, I also found the repository added automatically, at least
when you calling coveralls from the github with token.

Belive me, I threw a few days of my life to make it work from github action
:(((

пн, 8 мар. 2021 г., 15:54 Anton Vodonosov <notifications@github.com>:

**Comment by avodonosov** [comment]
> Yes. By the way, I also found the repository added automatically, at least
when you calling coveralls from the github with token.

Maybe this type of registration, via API call with token, prevents further use without token. Without token the auto-registration does not happen, as I remember - I've been experimenting with cl-coveralls for drakma, and until I registered the repo, it didn't work.

I worry that if you are mistaken about token being required by coveralls, the cl-coveralls code blocks this use case anyways, without need.

Why not let coveralls API validate the parameters and decide what is acceptable and what is not? The developer can then read the coveralls API error message in logs and fix the problem.

If the API error message is obscure and not informative, then OK, let's turn the `(error "Please, set COVERALLS_REPO_TOKEN env variable. It is required.")` into a `(warn "COVERALLS_REPO_TOKEN env variable is absent, the Coveralls API call may fail")`
